portal/static/js/initialQueries.js

```diff
@@ -0,0 +1,949 @@
+      if (callback) {
+        callback(data);
+      };
+    });
+  };
+
+  FieldsChecker.prototype.initSections = function() {
+    var self = this;
+    self.setSections();
+    if (!self.getConfig()) {
+        this.mainSections = this.defaultSections;
+    }
+    else {
+      var defaultSections = this.defaultSections;
+      for (var section in defaultSections) {
+          if (defaultSections[section].required) {
+            this.mainSections[section] = defaultSections[section];
+          } else if (self.inConfig(defaultSections[section].config)) {
+            this.mainSections[section] = defaultSections[section];
+        };
+      };
+    };
+  };
+
+  FieldsChecker.prototype.setSections = function() {
+      var preselectClinic = this.preselectClinic;
+      var self = this;
+      var i18next = this.__getDependency("i18next");
+      var orgTool = this.__getDependency("orgTool");
+      /*
+       * main sections blueprint object, this will help keeping track of missing fields for each section
+       *
+       */
+      this.defaultSections =  {
+          "topTerms": {
+                display: i18next.t("terms"),
+                config: "website_terms_of_use,subject_website_consent,privacy_policy",
+                subsections: {
+                  "termsCheckbox": {
+                      fields: [$("#topTerms [data-type='terms'][data-required='true']")]
+                  }
+                },
+                handleIncomplete: function() {
+                  $("#aboutForm").addClass("full-size");
+                  $("#topTerms").removeClass("hide-terms").show();
+                  if (window.performance) {
+                    if (performance.navigation.type === 1) {
+                      //page has been reloaded;
+                      var agreedCheckboxes = $("#topTerms .terms-label:visible i");
+                      if (agreedCheckboxes.length > 1) {
+                          $("#termsReminderCheckboxText").text(i18next.t("You must agree to the terms and conditions by checking the provided checkboxes."));
+                      };
+                      if (agreedCheckboxes.length === 0) {
+                        $("#termsText").addClass("agreed");
+                      }
+                      $("#termsReminderModal").modal("show");
+                    };
+                  };
+                  setTimeout(function() { disableHeaderFooterLinks(); }, 1000);
+                }
+            },
+            "demographicsContainer": {
+                display: i18next.t("your information"),
+                config: "name,dob,role",
+                subsections:{
+                "nameGroup": {
+                  fields: [$("#firstname"), $("#lastname")]
+                }
+                ,
+                "rolesGroup": {
+                  fields: [$("input[name='user_type']")]
+                }
+                ,
+                "bdGroup": {
+                    fields: [$("#month"), $("#date"), $("#year")]
```

Comment:
How do these ids still work if there are multiple date forms on the page?

Reply:
@ivan-c Ivan, the other date fields are assigned different ids in those cases (e.g. biopsy date fields)